realm/realm-library/src/androidTest/kotlin/io/realm/ManagedSetTester.kt

```diff
@@ -104,12 +110,26 @@ class ManagedSetTester<T : Any>(
     }
 
     override fun toArray() {
-        // TODO
+        val set = initAndAssert()
+
+        // Empty set
+        assertEquals(0, set.toArray().size)
+
+        // Set with some values
+        realm.executeTransaction {
+            set.addAll(initializedSet)
+            assertEquals(initializedSet.size, set.toArray().size)
```

Comment:
Wouldn't you also be able to test the actual values with `assertArrayEquals`?

Reply:
Nope, because the ordering might be different. For example, for the following set `{"HELLO", "BYE", null}` a `TreeSet` implementation generates an array that looks like `[null, ""HELLO", "BYE"]`, whereas our iterator returns `[null, "BYE", "HELLO"]`.